src/v/cluster/tx_gateway_frontend.cc

```diff
@@ -2151,6 +2151,57 @@ tx_gateway_frontend::get_all_transactions() {
       });
 }
 
+ss::future<result<tm_transaction, tx_errc>>
+tx_gateway_frontend::get_tx(kafka::transactional_id tid) {
+    auto shard = _shard_table.local().shard_for(model::tx_manager_ntp);
+
+    if (!shard.has_value()) {
+        vlog(txlog.warn, "can't find a shard for {}", model::tx_manager_ntp);
+        co_return tx_errc::shard_not_found;
+    }
+
+    co_return co_await container().invoke_on(
+      *shard,
+      _ssg,
+      [tid](tx_gateway_frontend& self)
+        -> ss::future<result<tm_transaction, tx_errc>> {
+          auto partition = self._partition_manager.local().get(
+            model::tx_manager_ntp);
+          if (!partition) {
+              vlog(
+                txlog.warn,
+                "can't get partition by {} ntp",
+                model::tx_manager_ntp);
+              co_return tx_errc::partition_not_found;
+          }
+
+          auto stm = partition->tm_stm();
+
+          if (!stm) {
+              vlog(
+                txlog.error,
+                "can't get tm stm of the {}' partition",
+                model::tx_manager_ntp);
+              co_return tx_errc::unknown_server_error;
```

Comment:
unknown_server_error doesn't seem right here

Reply:
Why. If we do not have tm_stm smth goes wrong. Maybe transaction is disabled.
So for me right way is return fatal error for client. After it user can check log

Reply:
This seems like a retryable error, don't think it can ever happen, if it does, the partition is probably on its way to another node, the client should just retry. As we have seen with transactions, unknown_server_error throws the client off (may be even aborts), so should be a last resort when retries won't help.